src/routes/webhooks.js

```diff
@@ -0,0 +1,201 @@
+    }
+    res.status(400).send({
+      msg: 'Not found.',
+    });
+  }
+);
+
+/* This is a PUT request that updates token offer. */
+router.put('/offer/token', updateTokenOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        tokenAddress: req.body._token ? req.body._token : offer.tokenAddress,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(400).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates chain id offer. */
+router.put('/offer/chain', updateChainIdOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        chainId: req.body._chainId ? req.body._chainId : offer.chainId,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(400).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates status offer. */
+router.put('/offer/status', updateStatusOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        isActive:
+          req.body._isActive === undefined
+            ? offer.isActive
+            : req.body._isActive,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(404).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates new offer. */
+router.put('/offer', updateOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  return res.status(200).send(null);
+});
+
+/* This is a PUT request that updates offer trade. */
+router.put('/offer/trade', updateTradeOfferValidator, async (req, res) => {
```

Comment:
> @tcoratger
> 
> How should we proceed with this [event](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtPool.sol#L83) and this [one](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtOffer.sol#L81) if we do not have every param needed to update the database? Few of the params only the frontend knows.
> 
> LogTrade event is here to declare a new offer which has just been accepted by Lisa and therefore a new associated order. The majority of the parameters will already be known from the front-end (offer = Gordon address, offerId, address of the token deposited, amount deposited). In fact, you just have to listen to the blockchain for this event, in the database we will already have all the information with the hash of the transaction as well. All you have to do is match the elements and fill in the orderId thanks to the event, corresponding to the proper transaction hash.

From the event listened it is being updated the `isComplete` param from `order` found by `_idTrade`

Reply:
This is the payload for the event listened

```
{
  _grinderyChainId: '5',
  _grinderyTransactionHash: '0x9ac9dc684b299e1f7e93de22ca13530ffe0ad818c7f735ceb9657ac37ddab000',
  _idOffer: '0xcb21b0170c84ded985587ebf1cf042acd5aba5407914a30a872f1f0b97048767',
  _amount: '1000000000000000000',
  _idTrade: '0x2b3501a436df4e007bba7b3cfa95501910a7605ea29d9f8d7ffb19ec3a70cf0c',
  _offerer: '0x62E45140ea73a2becfB964c797314EF7efDAEFBE',
  _token: '0x0000000000000000000000000000000000000000'
}
```

Reply:
@juniormp 

`isComplete` is different, it comes from the destination blockchain. Indeed, it is the element in the database that allows you to close an order by declaring it as paid. 

So it has nothing to do with the master blockchain. To do this, you must listen to the `LogOfferPaid` event on the liquidity wallet smart contract. For the moment this is not a problem, we only have one liquidity wallet contract per blockchain, Tim's. So you have to listen to this event on this one.

As soon as you have a new `LogOfferPaid` event, you capture the `tradeId`, you fetch it from the database and you set this order as completed.

The payload you show here indeed corresponds to a new order placed by Lisa (but not yet paid for by Gordon). With the listening of this event you will be able to take the hash of the corresponding transaction, look for it in the `orders` database and fill in the `orderId` = `tradeId` which is currently empty in the database (because Lisa left the page just after confirming on Metamask, its transaction not having been completed, we have not yet been able to capture the `orderId`).